rbi/annotations/shopify-money.rbi

```diff
@@ -0,0 +1,173 @@
+  sig { returns(BigDecimal) }
+  attr_reader :value
+
+  sig { returns(T.any(Money::Currency, Money::NullCurrency)) }
+  attr_reader :currency
+
+  sig do
+    params(
+      value: T.nilable(T.any(Money, Numeric, String)),
+      currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
+    )
+      .void
+  end
+  def initialize(value, currency); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def zero?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def nonzero?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def positive?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def negative?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(Integer) }
+  def to_i(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(Float) }
+  def to_f(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(Integer) }
+  def hash(*args, **_arg1, &block); end
+
+  class << self
+    sig { params(block: T.nilable(T.proc.params(config: Money::Config).void)).void }
+    def configure(&block); end
+
+    sig do
+      params(
+        value: T.nilable(T.any(Money, Numeric, String)),
+        currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
+      )
+        .returns(Money)
+    end
+    def new(value = 0, currency = nil); end
+
+    sig do
+      params(
+        subunits: T.nilable(T.any(Money, Numeric, String)),
+        currency_iso: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
+        format: Symbol,
+      )
+        .returns(Money)
+    end
+    def from_subunits(subunits, currency_iso, format: :iso4217); end
+
+    sig { params(money1: Money, money2: Money).returns(Rational) }
+    def rational(money1, money2); end
+
+    sig { returns(T.nilable(T.any(Money::Currency, Money::NullCurrency, String))) }
+    def current_currency; end
+
+    sig { params(currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String))).void }
+    def current_currency=(currency); end
+
+    sig do
+      params(
+        new_currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
+        block: T.nilable(T.proc.void),
+      )
+        .void
+    end
```

Comment:
```suggestion
    sig do
      params(
        new_currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
        block: T.nilable(T.proc.returns(T.untyped)),
      )
      .returns(T.untyped)
    end
```

Looks like the method yields to the block so `void` won't work here. https://github.com/Shopify/money/blob/2ebdb97174e743829542b3dafbf6cca42512d084/lib/money/money.rb#L74